test/common/scale_up_replicas_threescale.go

```
@@ -0,0 +1,152 @@
+	scaleUpReplicas  int64 = 3
+	name                   = "3scale"
+	namespace              = "redhat-rhmi-3scale"
+)
+
+func TestReplicasInThreescale(t *testing.T, ctx *TestingContext) {
+
+	createWorkspaceRetryInterval := time.Second * 20
+	createWorkspaceTimeout := time.Minute * 7
+
+	apim, err := getAPIManager(ctx.Client, t)
+	if err != nil {
+		t.Fatalf("failed to get APIManager : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, numberOfReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas to start : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, t, scaleUpReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of updated replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, scaleUpReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, t, numberOfReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas has been reset")
+	if err := checkNumberOfReplicasAgainstValue(apim, numberOfReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+}
+
+func getAPIManager(dynClient k8sclient.Client, t *testing.T) (threescalev1.APIManager, error) {
+	apim := &threescalev1.APIManager{}
+
+	if err := dynClient.Get(goctx.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, apim); err != nil {
+		t.Fatalf("Error : %v ", err)
+	}
+
+	return *apim, nil
+}
+
+func updateAPIManager(dynClient k8sclient.Client, t *testing.T, replicas int64) (threescalev1.APIManager, error) {
+
+	apim, err := getAPIManager(dynClient, t)
+	if err != nil {
+		t.Fatalf("Error : %v ", err)
```

Same as above, I would just return the error here.